### PR TITLE
[improve][doc] improve pulsar I/O document

### LINKED ISSUE #17555: [Doc] There are many broken links in Pulsar IO chapter
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

When I read the Pulsar IO document,  I found many links got 404 errors.

### What is your suggestion?

Find out and fix them.

### Any reference?

For example:

https://pulsar.apache.org/docs/io-overview/#set

image

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by labuladong**
Fixes #17555

### Motivation

At first, I'd like to fix the broken links in `Pulsar I/O` chapter. But I found that not only this chapter has this problem, so I open #17599 for further discussion.

In this PR, I replace the docker introduction with a reference link and improve some code snippets.

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by labuladong** [comment]
cc @Anonymitaet @momo-jun

File: site2/docs/io-quickstart.md [hunk 1]
```diff
@@ -473,18 +460,22 @@ This example uses the PostgreSQL 12 docker image to start a single-node PostgreS
...
    ```
 
-5. Create a PostgreSQL table _pulsar_postgres_jdbc_sink_.
+5. Login PostgreSQL with default username and password:
 
    ```bash
    psql -U postgres postgres
+   ```
+
+6. Input following commands to create `pulsar_postgres_jdbc_sink` table:
```
Comment (Anonymitaet) [inline site2/docs/io-quickstart.md:475]:
```suggestion
6. Create a `pulsar_postgres_jdbc_sink` table using the following command:
```

1. Show "action / purpose" first.
2. Below is one command, so remove "s" for "command".

**Review by Anonymitaet — COMMENTED** [review]
Thanks for your contribution! 
I've left some comments, PTAL. 
Feel free to correct me if my understanding is inaccurate and let me know if you have any concerns. 😊

File: site2/docs/io-quickstart.md [hunk 2]
```diff
@@ -473,18 +460,22 @@ This example uses the PostgreSQL 12 docker image to start a single-node PostgreS
...
 
-4. Access to PostgreSQL.
+4. Access to PostgreSQL container.
 
    ```bash
    docker exec -it pulsar-postgres /bin/bash
    ```
 
-5. Create a PostgreSQL table _pulsar_postgres_jdbc_sink_.
+5. Login PostgreSQL with default username and password:
```
Comment (momo-jun) [inline site2/docs/io-quickstart.md:469]:
```suggestion
5. Log in to PostgreSQL with the default username and password:
```

File: site2/docs/io-quickstart.md (hunk 1 above)
Comment (labuladong) [reply site2/docs/io-quickstart.md:475]: Got it, thanks.

**Comment by labuladong** [comment]
Thanks for your review, changes were addressed. @Anonymitaet @momo-jun

**Review by Anonymitaet — APPROVED** [review]

**Review by momo-jun — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]

**Pull request merged**